shell/platform/android/io/flutter/view/FlutterView.java

```diff
@@ -597,17 +602,28 @@ private void handlePlatformMessage(String name, String message, final int respon
         OnMessageListenerAsync asyncListener = mAsyncOnMessageListeners.get(name);
         if (asyncListener != null) {
             asyncListener.onMessage(this, message, new MessageResponse() {
-              @Override
-              public void send(String response) {
-                  nativeInvokePlatformMessageResponseCallback(mNativePlatformView, responseId, response);
-              }
+                @Override
+                public void send(String response) {
+                    nativeInvokePlatformMessageResponseCallback(mNativePlatformView, responseId, response);
+                }
             });
             return;
         }
 
         nativeInvokePlatformMessageResponseCallback(mNativePlatformView, responseId, null);
     }
 
+    private int mNextResponseId = 1;
+    private final Map<Integer, MessageReplyCallback> mPendingResponses = new HashMap<Integer, MessageReplyCallback>();
+
+    // Called by native to respond to a platform message that we sent.
+    @CalledByNative
+    private void handlePlatformMessageResponse(int responseId, String response) {
+        MessageReplyCallback callback = mPendingResponses.remove(responseId);
```

Comment:
Is this being invoked on the Dart UI thread?
If so, then the callback and the access to the mPendingResponse map may need to be queued to the Android main thread